packages/@webex/plugin-cc/test/unit/spec/services/task/index.ts

```diff
@@ -0,0 +1,283 @@
+
+    expect(taskEmitSpy).toHaveBeenCalledWith(TASK_EVENTS.TASK_MEDIA, mockTrack);
+  });
+
+  it('test updating the task data', async () => {
+    const newData = {
+      type: CC_EVENTS.AGENT_CONTACT_ASSIGNED,
+      agentId: "723a8ffb-a26e-496d-b14a-ff44fb83b64f",
+      eventTime: 1733211616959,
+      eventType: "RoutingMessage",
+      interaction: {},
+      interactionId: taskId,
+      orgId: "6ecef209-9a34-4ed1-a07a-7ddd1dbe925a",
+      trackingId: "575c0ec2-618c-42af-a61c-53aeb0a221ee",
+      mediaResourceId: '0ae913a4-c857-4705-8d49-76dd3dde75e4',
+      destAgentId: 'ebeb893b-ba67-4f36-8418-95c7492b28c2',
+      owner: '723a8ffb-a26e-496d-b14a-ff44fb83b64f',
+      queueMgr: 'aqm',
+    };
+
+    expect(task.data).toEqual(taskDataMock);
+
+    task.updateTaskData(newData);
+
+    expect(task.data).toEqual(newData);
+  });
+
+  it('should accept a task and answer call when using BROWSER login option', async () => {
+    const answerCallSpy = jest.spyOn(webCallingService, 'answerCall');
+    
+    await task.accept();
+
+    expect(navigator.mediaDevices.getUserMedia).toHaveBeenCalledWith({ audio: true });
+    expect(LocalMicrophoneStream).toHaveBeenCalledWith(mockStream);
+    expect(answerCallSpy).toHaveBeenCalledWith(expect.any(LocalMicrophoneStream), taskId);
+  });
+
+  it('should call accept API for Extension login option', async () => {
+    webCallingService.loginOption = LoginOption.EXTENSION
+    
+    await task.accept();
+
+    expect(contactMock.accept).toHaveBeenCalledWith({interactionId: taskId});
+  });
+
+  it('should handle errors in accept method', async () => {
+    const error = {
+      details: {
+        trackingId: '1234',
+        data: {
+          reason: 'Accept Failed',
+        },
+      },
+    };
+
+    jest.spyOn(webCallingService, 'answerCall').mockImplementation(() => { throw error });
+
+    await expect(task.accept()).rejects.toThrow(new Error(error.details.data.reason));
+    expect(getErrorDetailsSpy).toHaveBeenCalledWith(error, 'accept', CC_FILE);
+  });
+
+  it('should decline call using webCallingService', async () => {
+    const declineCallSpy = jest.spyOn(webCallingService, 'declineCall');
+    const offSpy = jest.spyOn(webCallingService, 'off');
+    
+    await task.decline();
+
+    expect(declineCallSpy).toHaveBeenCalledWith(taskId);
+    expect(offSpy).toHaveBeenCalledWith(CALL_EVENT_KEYS.REMOTE_MEDIA, offSpy.mock.calls[0][1]);
+  });
+
+  it('should handle errors in decline method', async () => {
+    const error = {
+      details: {
+        trackingId: '1234',
+        data: {
+          reason: 'Decline Failed',
+        },
+      },
+    };
+
+    jest.spyOn(webCallingService, 'declineCall').mockImplementation(() => { throw error });
+    await expect(task.decline()).rejects.toThrow(new Error(error.details.data.reason));
+    expect(getErrorDetailsSpy).toHaveBeenCalledWith(error, 'decline', CC_FILE);
+  }); 
+
+  it('should hold the task', async () => {
```

Comment:
A test scenario to cover manually as well as in UT:

What happens in our CC-SDK if we do not pass the media resource ID in both cases hold and resume?

Reply:
It fails to hold/resume... the call is continued as usual